qiskit/assembler/assemble_schedules.py

```diff
@@ -184,12 +184,12 @@ def _assemble_instructions(
                                    name=instruction.name)
 
         if isinstance(instruction, PulseInstruction):  # deprecated
-            instruction = Play(SamplePulse(name=name, samples=instruction.command.samples),
+            instruction = Play(Waveform(name=name, samples=instruction.command.samples),
                                instruction.channels[0], name=name)
 
-        if isinstance(instruction, Play) and isinstance(instruction.pulse, SamplePulse):
+        if isinstance(instruction, Play) and isinstance(instruction.pulse, Waveform):
```

Comment:
A user could still send a `SamplePulse`. Then what would happen? It wouldn't be added properly to the resulting qobj.
```suggestion
        if isinstance(instruction, Play) and isinstance(instruction.pulse, (SamplePulse, Waveform)):
```
This means you'll have to include `SamplePulse` in the imports, still.